### PR TITLE
MGMT-4494 Unit tests for reconciling day2 worker

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
This PR is based on the work already done by Sudarshan here https://github.com/openshift/assisted-service/pull/1514/commits/7601896f469af6e186c5492df18105d4ae7c1aea and fixes the failing test case by making necessary changes.
Ref: https://issues.redhat.com/browse/MGMT-5236

### CODE REVIEW

File: internal/controller/controllers/bmh_agent_controller_test.go [hunk 1]
```diff
@@ -426,4 +435,115 @@ var _ = Describe("bmac reconcile", func() {
...
+				updatedHost := &bmh_v1alpha1.BareMetalHost{}
+				err = c.Get(ctx, types.NamespacedName{Name: host.Name, Namespace: testNamespace}, updatedHost)
+				Expect(err).To(BeNil())
+				Expect(updatedHost.ObjectMeta.Annotations).To(HaveKey(BMH_HARDWARE_DETAILS_ANNOTATION))
+
+				spokeBMH := &bmh_v1alpha1.BareMetalHost{}
+				spokeClient := bmhr.spokeClient
+				err = spokeClient.Get(ctx, types.NamespacedName{Name: host.Name, Namespace: testNamespace}, spokeBMH)
+				Expect(err).To(BeNil())
+				Expect(spokeBMH.ObjectMeta.Annotations).To(HaveKey(BMH_HARDWARE_DETAILS_ANNOTATION))
```
Comment (rwsu) [inline internal/controller/controllers/bmh_agent_controller_test.go:532]: I would suggest going one step further and also assert that the contents are also the same.

`Expect(spokeBMH.ObjectMeta.Annotations[BMH_HARDWARE_DETAILS_ANNOTATION]).To(Equal(updatedHost.ObjectMeta.Annotations[BMH_HARDWARE_DETAILS_ANNOTATION]))
				`

**Review by rwsu — COMMENTED** [review]
Looks good. Just one suggestion.

File: internal/controller/controllers/bmh_agent_controller_test.go (hunk 1 above)
Comment (pawanpinjarkar) [reply internal/controller/controllers/bmh_agent_controller_test.go:532]: Done

**Review by rwsu — APPROVED** [review]

**Pull request merged**